drivers/ethernet/phy/Kconfig

```diff
@@ -33,6 +33,16 @@ config PHY_GENERIC_MII
 	  This is a generic MII PHY interface that communicates with the
 	  PHY using the MDIO bus.
 
+config PHY_ADIN1100
+	bool "ADIN1100 PHY driver"
+	default y
+	depends on DT_HAS_ADI_ADIN1100_PHY_ENABLED
+	depends on ETH_STM32_HAL
+	depends on MDIO
+	depends on MDIO_ST_STM32_HAL
```

Comment:
this should not be dependent on the implementations being stm32 hal, in principle the phy driver should be able to fit with any mdio controller or ethernet device

Reply:
done, thanks, removed both MDIO_ST_STM32_HAL and ETH_STM32_HAL, since stm32 mdio driver goes trough stm32 HAL.

Reply:
it seems they are still there, is it intentional